tests/arch/arm/arm_sw_vector_relay/src/arm_sw_vector_relay.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <linker/linker-defs.h>
+#include <syscall_handler.h>
+
+#include <arch/arm/aarch32/cortex_m/cmsis.h>
+
+extern uint32_t _vector_table;
+extern uint32_t __vector_relay_handler;
+extern uint32_t _vector_table_pointer;
+
+static uint32_t vector_table_addr = (uint32_t)&_vector_table;
+static uint32_t vector_relay_table_addr = (uint32_t)&__vector_relay_table;
+static uint32_t vector_relay_handler_func = (uint32_t)&__vector_relay_handler;
+
+void test_arm_sw_vector_relay(void)
+{
+	/* Verify that the vector relay table entries (except first two
+	 * entries for MSP and ResetHandler) point to the relay handling
+	 * function.
+	 */
+	const uint32_t *vector_relay_table_addr_val =
+		(uint32_t *)(vector_relay_table_addr);
+
+	for (int i = 2; i < 16 + CONFIG_NUM_IRQS; i++) {
+		zassert_true(vector_relay_table_addr_val[i] ==
+			vector_relay_handler_func,
+			"vector relay table not pointing to the relay handler: 0x%x, 0x%x\n",
+			vector_relay_table_addr_val[i],
+			vector_relay_handler_func);
+	}
+
+#if defined(CONFIG_CPU_CORTEX_M_HAS_VTOR)
+
+	/* Verify that the forwarding vector table and the real
+	 * interrupt vector table respect the VTOR.TBLOFF alignment
+	 * requirements.
+	 */
+	uint32_t mask = MAX(128, POW2_CEIL(4 * (16 + CONFIG_NUM_IRQS))) - 1;
+
+	zassert_true(((vector_table_addr) & mask) == 0,
+		"vector table not properly aligned: 0x%x\n",
+		vector_table_addr);
+	zassert_true(((vector_relay_table_addr) & mask) == 0,
+		"vector relay table not properly aligned: 0x%x\n",
+		vector_relay_table_addr);
+
+	/* Verify that the VTOR points to the real vector table,
+	 * NOT the table that contains the forwarding function.
+	 */
+	zassert_true(SCB->VTOR == (uint32_t)_vector_start,
+		"VTOR not pointing to the real vector table\n");
+#else
+	(void)vector_table_addr;
```

Comment:
Instead of doing this, you could simply make those three variables local to this function, what seems to be a bit more adequate, as they are used only here.

Reply:
done now. 
I guess, it felt better to have all variables declared/defined in a single place, when I started designing this test. Sure, they look better being locals.